TombEngine/Scripting/Internal/TEN/Effects/EffectsFunctions.cpp

```diff
@@ -511,12 +540,39 @@ namespace TEN::Scripting::Effects
 
 	/// Make an explosion. Does not hurt Lara
 	// @function MakeExplosion 
-	// @tparam Vec3 pos
-	// @tparam[opt=512] float size This will not be the size of the sprites, but rather the distance between the origin and any additional sprites.
-	// @tparam[opt=false] bool shockwave If true, create a very faint white shockwave which will not hurt Lara.
-	static void MakeExplosion(Vec3 pos, TypeOrNil<float> size, TypeOrNil<bool> shockwave)
+	// @tparam Vec3 pos World position.
+	// @tparam[opt=512] float size Size of the shockwave if enabled.
+	// @tparam[opt=false] bool shockwave If true, create a very faint shockwave which will not hurt Lara. For underwater rooms it creates a splash if pos is near the surface. Shockwave uses Main color if provided.
+	// @tparam[opt] Color maincol Main Color of the explosion and the shockwave. If not provided default explosion color will be used.
+	// @tparam[opt] Color secondCol Second Color of the explosion. If not provided default explosion color will be used.
```

Comment:
I don't fully get the meaning of `secondCol`. We have main color, and we have second color. It should either be main color and additional color, or primary color and secondary color. Also clarify the meaning - when primary and secondary colors are used?